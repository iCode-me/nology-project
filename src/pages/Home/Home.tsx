import Button from "../../components/Button/Button";
import Layout from "../../components/Layout/Layout";

// to see this page component action, run the app with nothing appended to the URL 🧐
export const Home = () => {
  return (
    <Layout>
      <h1>Page Heading</h1>
      <h2>Section Heading</h2>
      <h3>Panel Heading</h3>
      <p>
        Lorem ipsum dolor, paragraph text sit amet consectetur adipisicing elit.
        Non error dolor in nostrum minima odio a fuga saepe soluta adipisci
        perspiciatis maxime tempora.
      </p>
      <p>
        Lorem ipsum dolor, paragraph text sit amet consectetur adipisicing elit.
        Non error dolor in nostrum minima odio a fuga saepe soluta adipisci
        perspiciatis maxime tempora.
      </p>
      <Button
        label="Log to console"
        onClick={() => {
          console.log("button clicked");
        }}
      />
    </Layout>
  );
};

export default Home;
