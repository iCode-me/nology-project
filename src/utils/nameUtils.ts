export const formatName = (firstName: string, lastName: string): string => {
  return `${firstName} ${lastName}`;
};

// export const formatNameArray = (names: string[]): string => {
//   return names.join(" ");
// };
