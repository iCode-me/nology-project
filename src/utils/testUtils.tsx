import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

type CustomRenderOptions = {
  useRouting: boolean;
};

export const customRender = (
  ui: JSX.Element,
  options: CustomRenderOptions = {
    useRouting: true,
  }
) => {
  let uiResult = ui;

  if (options.useRouting) {
    uiResult = wrapWithRouting(ui);
  }

  return render(uiResult);
};

const wrapWithRouting = (ui: JSX.Element): JSX.Element => {
  return <BrowserRouter>{ui}</BrowserRouter>;
};
