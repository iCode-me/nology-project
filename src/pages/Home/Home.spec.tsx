import { customRender } from "../../utils/testUtils";
import Home from "./Home";

export default describe("Home Page", () => {
  it("Renders", () => {
    const { container } = customRender(<Home />);
    expect(container).toMatchSnapshot();
  });
});
