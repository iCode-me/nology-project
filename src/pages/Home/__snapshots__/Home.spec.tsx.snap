// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Renders 1`] = `
<div>
  <div
    class="layout"
  >
    <main
      class="layout__main"
    >
      <h1>
        Page Heading
      </h1>
      <h2>
        Section Heading
      </h2>
      <h3>
        Panel Heading
      </h3>
      <p>
        Lorem ipsum dolor, paragraph text sit amet consectetur adipisicing elit. Non error dolor in nostrum minima odio a fuga saepe soluta adipisci perspiciatis maxime tempora.
      </p>
      <p>
        Lorem ipsum dolor, paragraph text sit amet consectetur adipisicing elit. Non error dolor in nostrum minima odio a fuga saepe soluta adipisci perspiciatis maxime tempora.
      </p>
      <button
        class="button"
        type="button"
      >
        Log to console
      </button>
    </main>
  </div>
</div>
`;
